### ISSUE TITLE
.jscsrc incompatible with grunt-jscs 2.2 and/or jscs 2.4

### ISSUE THREAD

**Issue by iwehrman** [adobe-photoshop/spaces-design#3118]
A few hours ago, `grunt-jscs` 2.2.0 was released which incorporates jscs 2.4.0. Our package.json specifies grunt-jscs version `^2.1`, and 2.2.0 is compatible with this, so we were automatically upgraded. Unfortunately, it is clearly not backwards compatible because it triggers many additional errors. In particular it does not seem to like our `disallowTrailingWhitespaceInSource` extension:

```
Running "jscs:main" (jscs) task
Unsupported rule: disallowTrailingWhitespaceInSource at Gruntfile.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/application.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/dialog.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/documents.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/edit.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/example.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/export.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/export.js :
    95 |
    96 |        // helper to build play object for  individual layers
    97 |        var _buildPlayObject = function (layerID) {
---------------------------------------^
    98 |            var layerExports = documentExports.getLayerExports(layerID),
    99 |                layerExportsArray = layerExports.toJS(),
Expect valid jsdoc-block definition at src/js/actions/export.js :
   897 |        // In debug mode we should first do an abbreviated test
   898 |        // to see if the plugin is already running (perhaps on a remote generator connection)
   899 |        var _preCheck = function () {
--------------------------------^
   900 |            if (globalUtil.debug) {
   901 |                return _getWebSocketPort()
Expect valid jsdoc-block definition at src/js/actions/export.js :
   914 |
   915 |        // helper to enabled generator if necessary and then init exportService
   916 |        var _enableAndConnect = function () {
----------------------------------------^
   917 |            return _getGeneratorStatus()
   918 |                .then(function (enabled) {
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/guides.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/help.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/history.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/index.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/keyevents.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/layereffects.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Expect valid jsdoc-block definition at src/js/actions/layereffects.js :
   229 |     */
   230 |    var setAlpha = function (document, layers, effectType, effectIndex, alpha, coalesce) {
   231 |        var alphaUpdater = function (effect) {
-----------------------------------^
   232 |            if (effect && effect.color) {
   233 |                return { color: effect.color.setAlpha(alpha) };
Expect valid jsdoc-block definition at src/js/actions/layereffects.js :
   257 |    var setColor = function (document, layers, effectType, effectIndex, color, coalesce, ignoreAlpha) {
   258 |        if (ignoreAlpha) {
   259 |            var colorUpdater = function (effect) {
---------------------------------------^
   260 |                if (effect && effect.color) {
   261 |                    return { color: effect.color.setOpaque(color) };
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/layers.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/layers.js :
   263 |     */
   264 |    var _getLayerIDsForDocumentID = function (documentID) {
   265 |        var _getLayerIDs = function (doc) {
-----------------------------------^
   266 |            var docRef = documentLib.referenceBy.id(documentID),
   267 |                startIndex = (doc.hasBackgroundLayer ? 0 : 1),
Invalid jsdoc-block definition at src/js/actions/layers.js :
  1438 |    ungroupSelected.post = [_verifyLayerIndex, _verifyLayerSelection];
  1439 |
  1440 |    /**
------------^
  1441 |     * Changes the visibility of layer
  1442 |     *
Invalid jsdoc-block definition at src/js/actions/layers.js :
  1643 |    unlockSelectedInCurrentDocument.transfers = [setLocking];
  1644 |
  1645 |    /**
------------^
  1646 |     * Reset the layer z-index. Assumes that all layers are already in the model,
  1647 |     * though possibly out of order w.r.t. Photoshop's model.
Invalid jsdoc-block definition at src/js/actions/layers.js :
  1691 |    resetIndex.post = [_verifyLayerIndex, _verifyLayerSelection];
  1692 |
  1693 |    /**
------------^
  1694 |     * Moves the given layers to their given position
  1695 |     * In Photoshop images, targetIndex 0 means bottom of the document, and will throw if
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/libraries.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Expect valid jsdoc-block definition at src/js/actions/libraries.js :
  1191 |    var _toolModalStateChangedHandler;
  1192 |
  1193 |    var beforeStartup = function () {
--------------------------------^
  1194 |        var dependencies = {
  1195 |            // Photoshop on startup will grab the port of the CC Library process and expose it to us
Expect valid jsdoc-block definition at src/js/actions/libraries.js :
  1194 |        var dependencies = {
  1195 |            // Photoshop on startup will grab the port of the CC Library process and expose it to us
  1196 |            vulcanCall: function (requestType, requestPayload, responseType, callback) {
--------------------------------^
  1197 |                descriptor.getProperty("application", "designSpaceLibrariesInfo")
  1198 |                    .then(function (imsInfo) {
Expect valid jsdoc-block definition at src/js/actions/libraries.js :
  1205 |            // https://git.corp.adobe.com/pages/ProjectCentral/cc-libraries-api-js/tutorial-analytics_example.html
  1206 |            analytics: {
  1207 |                reportEvent: function (event, properties) {
-------------------------------------^
  1208 |                    // elementType: color, characterstyle, layerstyle, graphic
  1209 |                    // ("image" type is replaced with "graphic" to keep naming consistency)
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/mask.js :
     1 |/*
--------^
     2 | * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/menu.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/policy.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/preferences.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/sampler.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/search.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/search.js :
    58 |    toggleSearchBar.transfers = [dialog.openDialog, dialog.closeDialog];
    59 |
    60 |    var beforeStartup = function () {
--------------------------------^
    61 |        var searchStore = this.flux.store("search");
    62 |
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/search/documents.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/search/layers.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/search/libraries.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/search/menucommands.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/shapes.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/shortcuts.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/shortcuts.js :
   194 |            controller = this.controller;
   195 |
   196 |        var _getKeyDownHandlerForPhase = function (capture) {
-------------------------------------------------^
   197 |            return function (event) {
   198 |                // Disable shortcuts when the controller is inactive
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/superselect.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/tools.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/tools.js :
   806 |            }
   807 |
   808 |            var activateTool = function () {
---------------------------------------^
   809 |                var applicationStore = flux.store("application"),
   810 |                    currentDocument = applicationStore.getCurrentDocument();
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/transform.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Expect valid jsdoc-block definition at src/js/actions/transform.js :
  1124 |        _layerTransformHandler;
  1125 |
  1126 |    var beforeStartup = function () {
--------------------------------^
  1127 |        _artboardTransformHandler = synchronization.debounce(function () {
  1128 |            // After each action we call, document model changes
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/type.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/typetool.js :
     1 |/*
--------^
     2 | * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
     3 | *  
Unsupported rule: disallowTrailingWhitespaceInSource at src/js/actions/ui.js :
     1 |/*
--------^
     2 | * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
     3 | *
Expect valid jsdoc-block definition at src/js/actions/ui.js :
   609 |
   610 |        // Handles window resize for resetting superselect tool policies
   611 |        _resizeHandler = function (event) {
---------------------------------^
   612 |            windowResizeDebounced(event);
   613 |        };
>> 50 code style errors found!
```

CC @markelog

**Comment by iwehrman** [comment]
I pushed 170b49ece54798941a823ca0f2b07dd23bb5afef to master to work around this, but I'm leaving this issue open in order to get our .jscsrc file fixed up w.r.t. the new jscs release.

**Comment by iwehrman** [comment]
(Also, the adapter jscs version is woefully out of date.)

**Comment by markelog** [comment]
Fix for it, will be released tomorrow with 2.5 version

**Comment by iwehrman** [comment]
Thanks @markelog!

**Comment by markelog** [comment]
btw, check it out - http://jscs.info/overview#versioning-semver :)

**Comment by iwehrman** [comment]
Fixed by #3170.
